rts/src/main/java/eta/runtime/io/MemoryManager.java

```diff
@@ -237,4 +242,161 @@ public static long loadStringsUTF8(String[] ss) throws UnsupportedEncodingExcept
+        ByteBuffer dest = getBoundedBuffer(destAddress);
+        return copyByteBuffer(src,srcOffset,dest,destOffset,n);
+    }
+
+    
+    public static ByteBuffer copy( long srcAddress, int srcOffset
+                                 , long destAddress, int destOffset
+                                 , int size) {
+        ByteBuffer src  = getBoundedBuffer(srcAddress);
+        ByteBuffer dest = getBoundedBuffer(destAddress);
+        return copyByteBuffer(src,srcOffset,dest,destOffset,size);
+    }
+
+    public static ByteBuffer copy(long srcAddress, long destAddress, int size) {
+        return copy(srcAddress, 0, destAddress, 0, size);
+    }
+
+    public static ByteBuffer set(long address, int val, int size) {
+        ByteBuffer buffer = getBoundedBuffer(address);
+        while (size-- != 0) {
+            buffer.put((byte) val);
+        }
+        return buffer;
+    }
+       
+    public static ByteBuffer set(long address, byte[] bytes) {
+        ByteBuffer buf = getBoundedBuffer(address);
+        buf.put(bytes);
+        return buf;
+    }
+
+    public static long allocateAndSet(byte[] bytes) {
+        long address = allocateBuffer(bytes.length,true);
+        set(address,bytes);
+        return address;
+    }
+    
+    public static ByteBuffer move(long srcAddress, long destAddress, int size) {
+        ByteBuffer src  = getBoundedBuffer(srcAddress);
+        ByteBuffer dest = getBoundedBuffer(destAddress);
+        ByteBuffer copy = ByteBuffer.allocate(size);
+        src.limit(src.position() + size);
+        copy.put(src);
+        copy.flip();
+        dest.put(copy);
+        return dest;
+    }
+
+    public static int compare(long a1, long a2, int n)  {
+        ByteBuffer b1 = getBoundedBuffer(a1);
+        ByteBuffer b2 = getBoundedBuffer(a2);
+        return compare(b1, b2, n);
+    }
+
+    public static int compare(ByteBuffer b1, ByteBuffer b2, int n)  {
+        while (n-- != 0) {
+            int a = b1.get() & 0xFF;
+            int b = b2.get() & 0xFF;
+            if (a != b) {
+                return a - b;
+            }
+        }
+        return 0;
+    }
+
+    public static int compare(long a1, int o1, long a2, int o2, int n) {
+        return compare(getBoundedBuffer(a1,o1,n), getBoundedBuffer(a2,o2,n), n);
+    }
+
+    public static int compare(long a1, int o1, ByteBuffer b2, int n) {
+        return compare(getBoundedBuffer(a1,o1,n), b2, n);
+    }
+
+    public static int compare(ByteBuffer b1, long a2, int o2, int n) {
+        return compare(b1, getBoundedBuffer(a2,o2,n), n);
+    }
+
+    public static ByteBuffer chr(ByteBuffer b, int c, int n) {
+        c = (int)((byte) c);
+        b = b.duplicate();
+        int idx = chrIndex(b, c, n);
+        if (idx == -1)
+            return emptyBuffer;
+        else
+            return (ByteBuffer) b.position(b.position() - 1);
+    }
+
+    public static int chrIndex(ByteBuffer b, int c, int n) {
+        c = (int)((byte) c);
```

Comment:
Same comment as above - maybe the `int c` shoud be `byte c` to convey that only the low 8 bits will be used?